events/api/src/main/java/org/projectnessie/events/api/CommittingEvent.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.events.api;
+
+/**
+ * Event that is emitted when a transaction is written to the catalog.
+ *
+ * <p>This type has 3 child interfaces:
+ *
+ * <ul>
+ *   <li>{@link CommitEvent}: for commits;
+ *   <li>{@link MergeEvent}: for merges;
+ *   <li>{@link TransplantEvent}: for transplants.
+ * </ul>
+ */
+public interface CommittingEvent extends Event {
+
+  /**
+   * The source reference where the committed operations came from. This is usually a branch, but
+   * not always (e.g. it could be a tag or a detached reference).
+   *
+   * <p>For commits, this is always the same as the {@linkplain #getTargetBranch() target branch}.
+   */
+  String getSourceReference();
+
+  /** The target branch where the committed operations were applied to. */
+  String getTargetBranch();
```

Comment:
Should we use the full-ref-name here to allow for future changes?
For example, if we wanna have a separate reference type for transactions.

Reply:
You are right, but at this point, I think we need to introduce a separate `Reference` type. I will try that.

On a side note, I think I will also rename all occurrences of the word "Branch" as in this method with "Reference" which sounds more future-proof, wdyt? E.g. `getTargetReference()`.

Reply:
`getTargetReference()` SGTM